### PR TITLE
Shared events edits

### LINKED ISSUE #77: SharedEvents - Get more info on users
## SharedEvents - Get more info on users

REX Birse :
Some information about users seems to be missing on the user management page of a shared event:
- The user's balance
- The amount that has been or will be debited from the user's balance

### PR DESCRIPTION
**Pull request by rowsky**
## Changes
- SharedEventManageUsers: Show total ponderation price
- SharedEventManageUsers: Show user balance
- Fix typo
- SharedEventManageUsers: Fix permission for finished events
- SharedEventManageUsers: Show total ponderation price on finished events
- Hide some columns on mobile
- SharedEventCreate: Disable self registration deadline field when self registration is not allowed
- Edit flash messages css & content
- Adding specific xlsx messages

## Fixes
- Fixes #77

### CODE REVIEW

File: finances/templates/finances/js/sharedevent_create_form_logic.html [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  $("#id_allow_self_registeration").change(function() {
+    if (this.checked) {
+        $("#id_date_end_registration").prop('disabled', false);
+    } else {
+	    $("#id_date_end_registration").prop('disabled', true);
+	}
+  });
+});
+</script>
+{% endblock %}
```
Comment (Eyap53) [inline finances/templates/finances/js/sharedevent_create_form_logic.html]: This is full js, right ?  It should then go to the static folder :)

File: finances/templates/finances/sharedevent_create.html [hunk 2]
```diff
@@ -15,4 +15,6 @@
     </div>
 </div>
 
+{% include 'finances/js/sharedevent_create_form_logic.html' %}
```
Comment (Eyap53) [inline finances/templates/finances/sharedevent_create.html]: Idem: Can be static js

File: finances/templates/finances/sharedevent_finish.html [hunk 3]
```diff
@@ -20,12 +20,12 @@
         Terminer l'évènement {{ se }}
     </div>
     <div class="panel-body">
-        <p>
-            <strong>Rappel :</strong><br>
-             - Prix total: {% if total_price %}{{ total_price }}{% else %}Non renseigné{% endif %}<br>
-             - Nombre de part: {{ total_weights_participants }}<br>
-             {% if ponderation_price %}- Prix de revient par part: {{ ponderation_price }}<br>{% endif %}
-             <br>
+        <strong>Rappel :</strong><br>
```
Comment (Eyap53) [inline finances/templates/finances/sharedevent_finish.html]: To be w3c valid, it should be wrapped in a \<p>

File: finances/templates/finances/sharedevent_finish.html [hunk 4]
```diff
@@ -20,12 +20,12 @@
...
+        <ul>
+            <li>Prix total: {% if total_price %}{{ total_price }} €{% else %}Non renseigné{% endif %}</li>
+            <li>Nombre de part: {{ total_weights_participants }}</li>
+            {% if ponderation_price %}<li>Prix de revient par part: {{ ponderation_price }}</li>{% endif %}
+		</ul>
```
Comment (Eyap53) [inline finances/templates/finances/sharedevent_finish.html]: Nice !

File: finances/templates/finances/sharedevent_manage_users.html [hunk 5]
```diff
@@ -150,7 +173,7 @@
 
   {% include 'finances/js/ajax_ponderation.html' with list_weights=list_weights %}
   {% if price %}
-    {% if state == "participants" or state == "registrants" %}
+    {% if state == "participants" or state == "registrants" or state == "users" %}
```
Comment (Eyap53) [inline finances/templates/finances/sharedevent_manage_users.html]: Unless you changed the states, this if is now useless ?

File: finances/templates/finances/sharedevent_update.html [hunk 6]
```diff
@@ -60,10 +60,10 @@
     </div>
     <div class="panel-body">
       Nombre de pré-inscrits : <strong>{{ number_registrants }}</strong><br />
-      Nombre total de pré-inscription : <strong>{{ total_weights_registrants }}</strong><br />
+      Nombre total de pré-inscriptions : <strong>{{ total_weights_registrants }}</strong><br />
       <br />
       Nombre de participants : <strong>{{ number_participants }}</strong><br />
-      Nombre total de participation : <strong>{{ total_weights_participants }}</strong><br />
+      Nombre total de participations : <strong>{{ total_weights_participants }}</strong><br />
```
Comment (Eyap53) [inline finances/templates/finances/sharedevent_update.html]: 👍

File: finances/views.py [hunk 7]
```diff
@@ -1843,30 +1839,41 @@ def form_valid(self, form, *args, **kwargs):
...
+                  if user.count() > 0:
+                      user = User.objects.get(username=username)
+                      try:
+                          pond = int(row[1].value) # Should be an int. Else, raise an error
+                          if pond > 0:
+                              self.se.change_weight( user, pond, isParticipant )
+                      except:
+                          errors.append( "Erreur avec " + username + " (ligne n*" + str(i) + "). A priori pas ajouté." )
+                  else:
+                      errors.append( "L'utilisateur " + username + " n'existe pas. (ligne n*" + str(i) + ").")
```
Comment (Eyap53) [inline finances/views.py]: If user = User.objects.filter(username=username) doesn't find a user, i'm pretty sure it raise an error (catched by line 1865). Are you sure it's relevant ?

File: finances/views.py [hunk 8]
```diff
@@ -1843,30 +1839,41 @@ def form_valid(self, form, *args, **kwargs):
...
+                else:
+                  nb_empty_rows += 1
```
Comment (Eyap53) [inline finances/views.py]: If you don't add an error there, you are losing the line data

File: templates/base_sober.html [hunk 9]
```diff
@@ -122,7 +122,20 @@
...
+                                    {% if message.tags == 'info' %}fa-info-circle{% endif %}
+                                    {% if message.tags == 'warning' %}fa-exclamation-triangle{% endif %}
+                                    {% if message.tags == 'error' %}fa-exclamation-circle{% endif %}
+							      ">
+								  </i>
+								</div>
+								<div>
+                                   {{ message |linebreaks }}
+								</div>
+							  </div>
```
Comment (Eyap53) [inline templates/base_sober.html]: Not yet tried, but seems great !

File: templates/base_sober.html [hunk 10]
```diff
@@ -159,5 +172,17 @@
...
+  {% for message in messages %}
+    $(function() {
+      setTimeout(function() {
+          $(".message").hide('blind', {}, 500)
+      }, 25000);
+    });
+  {% endfor %}
+{% endif %}
+</script>
+
```
Comment (Eyap53) [inline templates/base_sober.html]: For clarity, can be moved within a templates/js/timer_flash_message.js file (or something similar) 😺

File: finances/templates/finances/js/sharedevent_create_form_logic.html (hunk 1 above)
Comment (rowsky) [reply finances/templates/finances/js/sharedevent_create_form_logic.html]: Well, you are right!

File: finances/views.py (hunk 7 above)
Comment (rowsky) [reply finances/views.py]: Nope, there is no error, when user is not found, the query result will be empty.
However, there is no need to store first that query into a variable.

File: finances/views.py (hunk 8 above)
Comment (rowsky) [reply finances/views.py]: On which line ?

Comment (Eyap53) [reply finances/views.py]: I just realized a manager can import a excel with all the users, and only add some weight for only a few of them. So my bad, it's fine like this

File: finances/views.py (hunk 7 above)
Comment (Eyap53) [reply finances/views.py]: Oh yeah, you are using filter and not get... Why that, btw ?

Comment (rowsky) [reply finances/views.py]: To check if a user exists before trying to add it, get generates an error, but not filter (as long as i'm using existing object attributes ...), and show a specific error when user is not found 😉

**Pull request merged**